tests/rptest/tests/schema_registry_test.py

```diff
@@ -1656,6 +1669,22 @@ def test_protobuf(self):
         assert result_raw.status_code == requests.codes.ok
         assert result_raw.json() == [2]
 
+    @cluster(num_nodes=3)
+    def test_restart(self):
+        """
+        Smoketest get_restart endpoint
+        """
+        result_raw = self._get_restart(auth=(self.username, self.password))
+        self.logger.debug(result_raw)
+        assert result_raw.json()['error_code'] == 40101
+
+        super_username, super_password, _ = self.redpanda.SUPERUSER_CREDENTIALS
+
+        self.logger.debug("Restart SR internal store")
+        result_raw = self._get_mode(auth=(super_username, super_password))
+        self.logger.debug(result_raw)
+        assert result_raw.status_code == requests.codes.ok
```

Comment:
The basic auth test here is actually failing. The logs show the following client errors after the client is restarted.
```
INFO  2022-12-14 18:40:14,565 [shard 0] pandaproxy - service.cc:182 - Schema registry successfully initialized
WARN  2022-12-14 18:40:14,585 [shard 0] pandaproxy - service.cc:229 - mitigate_error: kafka::client::partition_error ({topic_partition: {_schemas}:0}, { error_code: unknown_topic_or_partition [3] })
WARN  2022-12-14 18:40:14,633 [shard 0] pandaproxy - service.cc:229 - mitigate_error: kafka::client::partition_error ({topic_partition: {_schemas}:0}, { error_code: not_leader_for_partition [6] })
WARN  2022-12-14 18:40:14,720 [shard 0] pandaproxy - service.cc:229 - mitigate_error: kafka::client::partition_error ({topic_partition: {_schemas}:0}, { error_code: not_leader_for_partition [6] })
WARN  2022-12-14 18:40:14,887 [shard 0] pandaproxy - service.cc:229 - mitigate_error: kafka::client::partition_error ({topic_partition: {_schemas}:0}, { error_code: not_leader_for_partition [6] })
WARN  2022-12-14 18:40:15,221 [shard 0] pandaproxy - service.cc:229 - mitigate_error: kafka::client::partition_error ({topic_partition: {_schemas}:0}, { error_code: not_leader_for_partition [6] })
WARN  2022-12-14 18:40:15,897 [shard 0] pandaproxy - service.cc:229 - mitigate_error: kafka::client::partition_error ({topic_partition: {_schemas}:0}, { error_code: not_leader_for_partition [6] })
INFO  2022-12-14 18:40:16,768 [shard 0] raft - [group_id:1, {kafka/_schemas/0}] consensus.cc:1579 - Received vote request with larger term from node {id: {1}, revision: {25}}, received 1, current 0
INFO  2022-12-14 18:40:16,827 [shard 0] storage - segment.cc:641 - Creating new segment /var/lib/redpanda/data/kafka/_schemas/0_25/0-1-v1.log
WARN  2022-12-14 18:40:17,184 [shard 0] pandaproxy - service.cc:229 - mitigate_error: kafka::client::partition_error ({topic_partition: {_schemas}:0}, { error_code: not_leader_for_partition [6] })
```

Makes me wonder if I'm missing something in the client config or if the client expects the user/pass from the header (it should not though).

Reply:
Possible the client needs its internal state reset. Consider making it a unique_ptr then re initializing it when necessary.

Reply:
Oh, the other problem coudl be because I'm calling `get_mode` instead of `get_restart` :face-palm: